core/src/main/scala/ai/forestflow/utils/SourceStorageProtocols.scala

```diff
@@ -85,7 +93,44 @@ object SourceStorageProtocols extends StrictLogging {
   }
 
   case object S3 extends EnumVal {
-    def downloadDirectoryImpl(remotePath: String, localDirectory: File, fqrv: FQRV, sslVerify: Boolean): Unit = ???
+
+    private implicit val actorSystem: ActorSystem = ActorSystem("Application-s3-stream")
+    private implicit val executionContext: ExecutionContext = actorSystem.dispatcher
```

Comment:
I also believe we have an execution context defined somewhere else. In either case, using the system's dispatcher is a very bad idea for blocking code or code that has the potential to run long and has the potential to stall the actor system. There are already dispatchers defined for code that has blocking behavior. See how you can utilize those or if something needs to be re-considered to pass along that information.